include/LightGBM/utils/common.h

```diff
@@ -604,6 +610,11 @@ inline void ObtainMinMaxSum(const float *w, int nw, float *mi, float *ma, double
   if (su != nullptr) *su = sumw;
 }
 
+template <typename T>
+inline void ObtainMinMaxSum(const T *w, int nw, T *mi, std::nullptr_t, double *su) {
```

Comment:
```su``` could be ```T2``` type. 
I prefer to remove this function, and using ```ObtainMinMaxSum<float, double>(w, nw, mi, (float*)nullptr, su)``` to call the function.